src/java.base/share/native/launcher/main.c

```diff
@@ -34,6 +34,18 @@
 #include "jli_util.h"
 #include "jni.h"
 
+#ifdef UNDEFINED_BEHAVIOR_SANITIZER
+// Override weak symbol exposed by UBSan to override default options. This is called by UBSan
+// extremely early during library loading, before main is called. We need to override the default
+// options because by default UBSan only prints a warning for each occurrence. We want jtreg tests
+// to fail when undefined behavior is encountered. We also want a full stack trace for the offending
+// thread so it is easier to track down. You can override these options by setting the environment
+// variable UBSAN_OPTIONS.
+JNIEXPORT const char* __ubsan_default_options() {
+  return "halt_on_error=1,print_stacktrace=1";
+}
+#endif // UNDEFINED_BEHAVIOR_SANITIZER
+
 /*
```

Comment:
As I said we have more launcher than 'java', if you put this method here you must put it in all launchers.
I.e. all binaries the call JNI_CreateJavaVM, such out tests, such as jni/daemonDestroy/TestDaemonDestroy.java

Reply:
Attempting an approach that automatically includes `__ubsan_default_options` in binaries by "automagically" including a source file for anything using `SetupNativeCompilation` with `TYPE` being `EXECUTABLE`.

Reply:
Okay, I was able to get the autoconf magic working. Instead of having to copy-paste `__ubsan_default_options` to every launcher, it is instead auto-inserted in `SetupNativeCompilation` by including a source file which exports it for anything that is `EXECUTABLE`.